irestweaks.php

```diff
@@ -5,6 +5,44 @@
 use CRM_Irestweaks_ExtensionUtil as E;
 // phpcs:enable
 
+/**
+ * Implements hook_civicrm_postProcess().
+ *
+ * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_postProcess
+ */
+function irestweaks_civicrm_postProcess($formName, &$form) {
+  // Detect contribution page and contribution confirm page
+  if ($formName === 'CRM_Contribute_Form_Contribution_Confirm') {
+    // Get the source param by the entryURL in the controler array
+    $controller = $form->getVar('controller');
+    // Process to get the source param
+    $params = explode('?', $controller->_entryURL);
+    parse_str(end($params), $parseURL);
+
+    $getSource = '';
+
+    // Remove amp; since it was not remove using parse_str
+    foreach ($parseURL as $key => $value) {
+      $newKey = str_replace('amp;', '', $key);
+
+      if ($newKey === 'source') {
+        $getSource = 'Online Contribution: ' . $form->_values['title'] . ' - ' . $value;
```

Comment:
This line guesses that civicrm always uses a Source field value like `'Online Contribution: ' . $form->_values['title']`, but that may not be true in all situations; or it may be different in some civicrm versions.

Instead, this function needs to get the existing Source field value as `$existingSource` and then modify it as `$existingSource . " - $value"`.  To get the existing Source field value, please use the `contribution.get` api, passing it the `id` parameter so it knows which contribution to get.